asinfo.c

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (c) 2014-2020 by Farsight Security, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* asprintf() does not appear on linux without this */
+#define _GNU_SOURCE
+
+#include <arpa/inet.h>
+#include <arpa/nameser.h>
+#include <netinet/in.h>
+#include <errno.h>
+#include <netdb.h>
+#include <resolv.h>
+#include <stdlib.h>
+#include <string.h>
+#include "asinfo.h"
+#include "globals.h"
+
+static struct __res_state res;
+
+static const char *asinfo_from_ipv4(const char *, char **, char **);
+#ifdef asinfo_ipv6
```

Comment:
explain why we need asinfo_ipv6.

Reply:
see also:

To github.com:dnsdb/dnsdbq.git
   1fbef16..15c743d  asnlookup -> asnlookup